src/Catrobat/Commands/RestoreBackupCommand.php

```diff
@@ -7,165 +7,58 @@
-        ],
-        ['timeout' => 86400], 'Extract files to server resources directory', $output
-      );
-      CommandHelper::executeShellCommand(
-        [
-          'sshpass', '-p', $backup_host_password, 'ssh', $backup_host_user.'@'.$backup_host_name,
-          'mysql', '-u', $backup_database_user, '-p'.$backup_database_password, $backup_database_name,
-          '<', $backup_host_directory.$backup_host_resource_directory.'database.sql" ',
-        ],
-        ['timeout' => 86400], 'Restore SQL file', $output
-      );
-      CommandHelper::executeShellCommand(
-        [
-          'sshpass', '-p', $backup_host_password, 'ssh', $backup_host_user.'@'.$backup_host_name,
-          'unlink', $backup_host_directory.$backup_host_resource_directory.'database.sql',
-        ],
-        ['timeout' => 86400], 'Remove files from server resources directory', $output
-      );
-      CommandHelper::executeShellCommand(
-        [
-          'sshpass', '-p', $backup_host_password, 'ssh', $backup_host_user.'@'.$backup_host_name,
-          'mysql', '-u', $backup_database_user, '-p'.$backup_database_password, $backup_database_name,
-          '-e', 'UPDATE program p SET p.apk_status = '.Program::APK_NONE.' WHERE p.apk_status != '.Program::APK_NONE.';',
-        ],
-        ['timeout' => 86400], 'Reset the apk status', $output
-      );
-      CommandHelper::executeShellCommand(
-        [
-          'sshpass', '-p', $backup_host_password, 'ssh', $backup_host_user.'@'.$backup_host_name,
-          'mysql', '-u', $backup_database_user, '-p'.$backup_database_password, $backup_database_name,
-          '-e', 'UPDATE program p SET p.directory_hash = \"null\" WHERE p.directory_hash != \"null\"',
-        ],
-        ['timeout' => 86400], 'Reset the directory hash', $output
-      );
-    }
-
-    $this->output->writeln('Import finished!');
-
-    return 0;
+    $output->writeln('Restore backup on localhost');
+
+    CommandHelper::executeShellCommand(
+            ['bin/console', 'catrobat:purge', '--force'], [], 'Purging database', $output
+        );
+
+    CommandHelper::executeShellCommand(
+            ['sh', 'bin/borg_restore_share.sh'], [86400],
+            'Executing borg restore script [timeout = 24h]', $output
+        );
+
+    $query = $this->entity_manager
+      ->createQuery('UPDATE App\Entity\Program p SET p.apk_status = :status WHERE p.apk_status != :status')
+        ;
+    $query->setParameter('status', Program::APK_NONE);
+    $result = $query->getSingleScalarResult();
+    $output->writeln('Reset the apk status of '.$result.' projects');
+
+    $query = $this->entity_manager
+      ->createQuery('UPDATE App\Entity\Program p SET p.directory_hash = :hash WHERE p.directory_hash != :hash')
+        ;
+    $query->setParameter('hash', 'null');
+    $result = $query->getSingleScalarResult();
+    $output->writeln('Reset the directory hash of '.$result.' projects');
```

Comment:
Since BORG Backup is quite mighty I dont think we need to reset the apk status and the directory _hash of a given program since those should not change with the new backup method and the apk are still existing.